### ISSUE TITLE
Uninstall never completes because it can't delete AMI

### ISSUE THREAD

**Issue by csrwng** [openshift/installer#1339]
# Version

```console
$ openshift-install version
openshift-install unreleased-master-457-g9596c7ffa001989c5453d8aff334f464b05c7a4d
```

# Platform (aws|libvirt|openstack):

aws

# What happened?

`openshift-install destroy cluster` never completes due to:

```
time="2019-02-28T16:36:25Z" level=debug msg="search for and delete matching resources by tag in us-east-1 matching aws.Filter{\"kubernetes.io/cluster/c3d1b23c-9718-40a5-p8tfs\":\"owned\"}"
time="2019-02-28T16:36:25Z" level=debug msg="deleting arn:aws:ec2:us-east-1:460538899914:image/ami-0f0b3ee1921278ed0: unrecognized EC2 resource type image"
```

# What you expected to happen?

The cluster destroy to complete successfully

# How to reproduce it (as minimally and precisely as possible)?

```console
$ openshift-install destroy cluster
```

# Anything else we need to know?

Introduced with 
https://github.com/openshift/installer/pull/1296

**Comment by dgoodwin** [comment]
@abhinavdahiya bit of a blocker on hive side, our CI can't pass if uninstall fails.

**Comment by wking** [comment]
Fixed with #1340 (replacing a `destroy` failure with a pre-check failure, which is at least more convenient to clean up).

/close
